src/main/java/com/tabnine/prediction/TabNineCompletion.java

```diff
@@ -55,17 +56,22 @@ public CompletionOrigin getOrigin() {
   }
 
   public String getSuffix() {
+    if (fullSuffix != null) {
+      return fullSuffix;
+    }
+
     String itemText = this.newPrefix + this.newSuffix;
     String prefix = this.oldPrefix;
     if (prefix.isEmpty()) {
-      return itemText;
+      return fullSuffix = itemText;
     }
 
     FList<TextRange> fragments = LookupCellRenderer.getMatchingFragments(prefix, itemText);
     if (fragments != null && !fragments.isEmpty()) {
       List<TextRange> list = new ArrayList<>(fragments);
-      return itemText.substring(list.get(list.size() - 1).getEndOffset());
+      return fullSuffix = itemText.substring(list.get(list.size() - 1).getEndOffset());
```

Comment:
what does `return fullSuffix = ...` do? this is unclear, please extract return value if this is intentional

Reply:
also below `return fullSuffix = "";`

Reply:
Cacheing
![image](https://user-images.githubusercontent.com/69103343/170954344-4a1b628b-0a9b-4b61-9a93-cde4e6e8cbbf.png)